panther_battery/src/adc_node.py

```diff
@@ -186,8 +205,42 @@ def _publish_battery_msg(
         else:
             battery_msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
 
+        # check battery health
+        with self._lock:
+            error_msg = None
+            if V_bat_mean < self.V_BAT_FATAL_MIN:
+                battery_msg.power_supply_health = BatteryState.POWER_SUPPLY_HEALTH_DEAD
+                error_msg = 'Battery voltage is critically low!'
+            elif V_bat_mean > self.V_BAT_FATAL_MAX:
+                battery_msg.power_supply_health = BatteryState.POWER_SUPPLY_HEALTH_OVERVOLTAGE
+                error_msg = 'Battery overvoltage!'
+            elif temp_bat >= self._high_bat_temp:
+                battery_msg.power_supply_health = BatteryState.POWER_SUPPLY_HEALTH_OVERHEAT
+                error_msg = 'Battery temperature is dangerously high!'
```

Comment:
```suggestion
                error_msg = 'Battery is overheating!'
```
For some reason, "dangerously" sounds strange in this case. IDK